tests/test_git_push.rs

```diff
@@ -179,6 +211,20 @@ fn test_git_push_unsnapshotted_change() {
     test_env.jj_cmd_success(&workspace_root, &["git", "push", "--change", "@"]);
     std::fs::write(workspace_root.join("file"), "modified").unwrap();
     test_env.jj_cmd_success(&workspace_root, &["git", "push", "--change", "@"]);
+    // test pushing two changes at once
+    std::fs::write(workspace_root.join("file"), "modified2").unwrap();
+    test_env.jj_cmd_success(&workspace_root, &["new", "-m", "bar"]);
+    std::fs::write(workspace_root.join("file"), "modified3").unwrap();
+    test_env.jj_cmd_success(
+        &workspace_root,
+        &["git", "push", "--change", "@", "--change", "@-"],
+    );
+    // specifying the same change twice doesn't break things
+    std::fs::write(workspace_root.join("file"), "modified4").unwrap();
+    test_env.jj_cmd_success(
+        &workspace_root,
+        &["git", "push", "--change", "@", "--change", "@"],
+    );
```

Comment:
Sorry I didn't notice it before, but this test was for testing a bug fixed in #418. It looks like there were no tests of the `--change` flag before, so please move the new tests to a new function.

Reply:
Well, I fixed this, but that revealed what seems to be a bug in my code. See the other comment I added.